### ISSUE TITLE
After Atomic Player closes, then some subscribed events emited

### ISSUE THREAD

**Issue by rsredsq** [AtomicGameEngine/AtomicGameEngine#248]
After I close Atomic Player, and if I subscribed to any render/update events, it's delte scene, node in JSComponent, and then event ecxecuted one more time(at this time scene, node are nulls). Here's an example:

``` JavaScript
"atomic component";

var component = function(self) {

    self.start = function() {
      var debug = self.scene.createComponent("DebugRenderer");

      self.subscribeToEvent("PostRenderUpdate", self.handlePostRenderUpdate);
    }

    self.handlePostRenderUpdate = function() {
         //Without the next line, it doesn't work and it throws an error
         //if(!self.scene) return;
         var world = self.scene.getComponent("PhysicsWorld2D");
         world.drawDebugGeometry();
    }

}

exports.component = component;
```

**Comment by JoshEngebretson** [comment]
Thanks for the report and example script, adding to Build 1 milestone, first step is to replicate

**Comment by JimMarlowe** [comment]
I am able to replicate the bad behavior by adding the submitted code to the Basic2D example ( and changing the example's PhysicsWorld to a PhysicWorld2D... ). 
What I found is, when the application checks for exiting via the engine and will call RunFrame() if there is no exit asserted. Sadly, the engine gets the exit message while the frame is in progress, and continues to call Update() before the application knows it's dead. I added exit guards to Update() and Render() and this stops the user code from crashing, since it will not receive the erroneous Update event.
